### PR TITLE
Add textobjects queries

### PR DESCRIPTION
**Pull request by ConradIrwin**
Co-Authored-By: Max <max@zed.dev>

Release Notes:

- vim: Added motions `[[`, `[]`, `]]`, `][` for navigating by section, `[m`, `]m`, `[M`, `]M` for navigating by method, and `[*`, `]*`, `[/`, `]/` for comments. These currently only work for languages built in to Zed, as they are powered by new tree-sitter queries.
- vim: Added new text objects: `ic`, `ac` for inside/around classes, `if`,`af` for functions/methods, and `g c` for comments. These currently only work for languages built in to Zed, as they are powered by new tree-sitter queries.

### CODE REVIEW

File: crates/collab/src/api/events.rs [hunk 1]
```diff
@@ -1555,15 +1555,12 @@ fn for_snowflake(
...
             map.insert("signed_in".to_string(), event.signed_in.into());
             if let Some(country_code) = country_code.as_ref() {
-                map.insert("country_code".to_string(), country_code.clone().into());
+                map.insert("country".to_string(), country_code.clone().into());
             }
         }
 
         let user_properties = Some(serde_json::json!({
             "is_staff": body.is_staff,
-            "Country": country_code.clone(),
```
Comment (maxdeviant) [inline crates/collab/src/api/events.rs:1564]: Were these telemetry changes intentional?

Comment (ConradIrwin) [reply crates/collab/src/api/events.rs:1564]: Not on this branch

**Comment by opsnull** [comment]
Will these convenient grammatical movement features be supported in non-vim modes?

**Comment by ConradIrwin** [comment]
I think they should be, but it's not clear to me the best way to make them available. Do you use an editor that supports this, and if so, what does the UX feel like? (and if not, what would you like?)

**Comment by opsnull** [comment]
> I think they should be, but it's not clear to me the best way to make them available. Do you use an editor that supports this, and if so, what does the UX feel like? (and if not, what would you like?)

Yes, I'm using emacs, and emacs support lots of sexp selection/movement and mark.

eg.

``` text
| C-M-f   | forward-sexp               | Match forward token (e.g. string, parentheses) on right of cursor |
| C-M-b   | backward-sexp              | Match backward token on left of cursor |
| C-M-a   | beginning-of-defun         | Move cursor backward to start of function definition |
| C-M-e   | end-of-defun               | Move cursor forward to end of function definition |
| C-M-n   | forward-list               | Move forward by parentheses |
| C-M-p   | backward-list              | Move backward by parentheses |
| C-M-u   | backward-up-list           | Jump to upper level of current parentheses |
| C-M-@   | mark-sexp                  | Mark current S-expression |
| C-M-SPC | mark-sexp                  | Mark current S-expression |
| C-M-h   | mark-defun                 | Mark function |
| C-M-q   | prog-indent-sexp           | Format current S-expression |
```

**Pull request merged**